specs/configuration_files_specification.md

```diff
@@ -55,11 +55,13 @@ For example, an empty `/etc/foo/bar.conf` means that `/usr/lib/foo/bar.conf` is
 masked and thus not parsed.
 
 ## Drop-ins
-All configuration paths must support drop-ins. Supporting drop-ins means that in
-addition to parsing a full configuration file, an implementation also must parse all
-files in the drop-in directory associated with it, and merge the result in lexicographic
-order. Drop-ins are not full replacements, but must allow specifying a subset of settings
-present in the main configuration files.
+All configuration paths must support drop-ins.
+Supporting drop-ins means that in addition to parsing a full configuration file,
+an implementation also must parse all files in the drop-in directory associated with it,
+and merge the result in lexicographic order.
```

Comment:
if you mention the order here I think you should also say that drop-ins are supposed to override relevant settings in the file they extend

Reply:
also, lexicographic order is one thing, it also matters to say that settings in the drop-ins are supposed to override earlier settings in early drop-ins

Reply:
> if you mention the order here I think you should also say that drop-ins are supposed to override relevant settings in the file they extend

This is covered in the second paragraph below.

> also, lexicographic order is one thing, it also matters to say that settings in the drop-ins are supposed to override earlier settings in early drop-ins

I rewrote the whole section to be more explicit about the sort order. Also added an example with config files under different hierarchies, which is more interesting.